torch/nn/modules/transformer.py

```diff
@@ -863,9 +889,9 @@ def _get_activation_fn(activation: str) -> Callable[[Tensor], Tensor]:
 
 
 def _detect_is_causal_mask(
-        mask: Optional[Tensor],
-        is_causal: Optional[bool],
-        size: Optional[int]
+        mask: Optional[Tensor]=None,
+        is_causal: Optional[bool]=None,
+        size: Optional[int]=None,
```

Comment:
Not sure how picky you are about code style, but there should be spaces around the equals signs here.

Reply:
Also, at least the `mask` argument should not be `None` by default, right? Since that one should always be supplied.

Reply:
I think it can be because we're not checking whether the mask is None (which is a legit call to forward).  So, then we need to check this condition at all call sites, or allow None.  (Whether it should default to None is a valid ... )